### PR TITLE
fix: Break ties in the complexity-stats top-10 lists by path

### LINKED ISSUE #426: assess: top-10 lists in complexity-stats.json have no tie-break, so tied files swap between runs
## Problem Statement

The three top-10 lists in `complexity-stats.json` are sorted on a single key with no tie-break, so files with equal scores keep the order the scanning tools emitted them in. Tied files can swap in and out of a top-10 list between runs on the same commit.

## Technical Context

Observed on plugin 1.87.0 at commit `4890b69`. Paths are relative to the repository root.

`skills/assess/scripts/complexity-treemap.py`, lines 1073-1075:

```python
by_score = sorted(enriched, key=lambda f: -f["_score"])
by_ccn = sorted(enriched, key=lambda f: -f["ccn"])
by_loc = sorted(enriched, key=lambda f: -f["loc"])
```

`sorted` is stable, so ties keep input order, and input order comes from lizard and scc output, which is not sorted by path. The lists are written at lines 1158-1160 as `top_hotspots`, `top_complex` and `top_large`.

## Impact

- `top_hotspots` decides which files get a wiki hotspot page, a first-flagged date and a place in the attention ranking; a tie at the tenth position makes that membership depend on tool output order.
- Output for the same commit is not byte-identical across machines or tool versions, which weakens every golden test and any artefact built from these lists.
- The attention list received a deterministic tie-break in #357; these lists did not.

## Root Cause

Single-key sorts on values that tie often (many files share a cyclomatic complexity or a line count). Confidence: high.

## Solution Requirements

- Each of the three sorts breaks ties by repository-relative path, ascending.
- Any new per-file list written to the stats file is sorted by path.
- No change to which files rank where when scores differ.

## Success Criteria

- A fixture with twelve files of identical complexity and size: `top_complex` and `top_large` hold the first ten by path, in path order, regardless of the order the files are passed in.
- Two runs over the same fixture with the input order reversed produce byte-identical top-10 lists.

### PR DESCRIPTION
**Pull request by bjcoombs**
## Summary

The three top-10 lists in `.assess/complexity-stats.json` (`top_hotspots`, `top_complex`, `top_large`) sorted on a single key. Python's `sorted` is stable, so tied files kept the order lizard and scc emitted them in, and that order is not sorted by path. A file tied at the tenth position could swap in and out between runs on the same commit, on the same machine.

Each sort now breaks ties on the repository-relative path, ascending. Paths below are relative to the repository root.

Closes #426

## Changes Made

- `skills/assess/scripts/complexity-treemap.py`: the three `sorted` calls in `write_stats` take a tuple key, `(-primary, path)`, in place of the bare negated primary. Primary keys are unchanged: `-_score`, `-ccn`, `-loc`.
- Same file: the `write_stats` docstring now records the tie-break and why it exists.
- `skills/assess/tests/test_complexity_treemap.py`: three tests, covering first-ten-by-path, input-order independence, and primary-key precedence against a dominant file whose path sorts last.
- Same file, same defect class, outside the issue's three lists and found in review: three more selections ordered on a single key. `_hint_if_largest_files_scc_only` and `render`'s biggest-files listing now share one helper, `_largest_first`, which takes the n largest by a caller-supplied size and breaks ties on the path; they differed only in what they do when a path carries no token estimate, so that fallback is what gets passed in. `_warn_if_dominated_by_one_file` used `max`, which returns the first maximal element, and now orders on `(-loc, path)`.
- `.claude-plugin/plugin.json`: set against `main` at the time of the last push.

## Technical Details

Membership of `top_hotspots` decides which files get a wiki hotspot page under `.assess/hotspots/`, a first-flagged date, and a place in the attention ranking, so an unstable tenth position churns the wiki. The attention list took the same `(-primary, ..., path)` shape in #357; this matches that idiom rather than inventing a second one.

`STATS_SCHEMA_VERSION` stays at 5. The change adds and removes no key, and `assess_core._diff_is_reliable` compares two runs only when they share the layout version, so an unnecessary bump would discard every repository's cross-run diff on the first run after the upgrade.

An earlier revision also pinned `STATS_SCHEMA_VERSION` at exactly 5; that pin is gone, because the file already asserts a floor of 5 and an equality pin inside a tie-break test would fail on the next legitimate layout change. The reason the tie-break does not move the version now sits in the `write_stats` docstring.

A deterministic tie-break makes alphabetically-early paths win the tenth slot systematically, so on a repository with a large tied group at the boundary `src/a*` files hold hotspot pages that `src/z*` files never get. That bias is inherent to any total order and is the intended trade against the present coin-flip, where the tenth slot follows scanner emission order and can change between identical runs. It is recorded here as a known property.

The issue's second Solution Requirement, that any new per-file list is sorted by path, is discharged by adding none. The top-level arrays of per-file objects remain `excluded_generated`, `top_complex`, `top_hotspots`, `top_large`.

Tie fixtures need care: lizard de-duplicates byte-identical sources and leaves the duplicates to scc, which scores them `ccn 0` while the surviving file scores 1, so twelve identical files never tie. The test fixture gives each file a distinct function name of identical length. It also uses unpadded names (`f1.py` to `f12.py`), so byte order is `f1, f10, f11, f12, f2, ...` and `f8`/`f9` fall outside the first ten - an outcome neither creation order, numeric order nor directory order can produce.

## Testing

- `skills/assess`: 1790 passed.
- Root `tests/`: 623 passed, 8 skipped.
- `scripts/`: 275 passed.
- `ruff check` on `skills/assess` and `scripts`, and `mypy` on the deterministic core: clean.
- Behavioural check on a scratch fixture of twelve files tied on every key: `top_hotspots`, `top_complex` and `top_large` each read `src/f1.py,src/f10.py,src/f11.py,src/f12.py,src/f2.py,src/f3.py,src/f4.py,src/f5.py,src/f6.py,src/f7.py`. Before the change the same fixture produced `src/f5.py,src/f1.py,src/f4.py,src/f12.py,src/f9.py,src/f11.py,src/f8.py,src/f10.py,src/f3.py,src/f7.py`.
- Ranking where values differ is unchanged, now covered by a committed test: a fixture with one dominant file (ccn 41, loc 82) plus twelve tied files puts the dominant file first in all three lists even though its name sorts last. No prior `top_complex` or `top_large` assertion in the file had a winner that sorts after its rival by path, so precedence for those two lists was unguarded.
- Calling `write_stats` twice over the same twelve tied rows, once with the list as built and once reversed, now yields identical lists; before the change the two runs echoed their input orders.

The two stderr selections matter for the same reason the lists do, though nothing they produce reaches the sidecar. The scc-only hint inspects the five largest files and goes quiet if any one of them is code; at a tied boundary a non-qualifying file could enter that set on one run and not the next, so the hint fired intermittently on an unchanged repository. The dominance warning names a single suspect file, and two files of equal LOC made that name follow scanner emission order - reachable, because the threshold is 30 percent of total LOC, not a majority. `render`'s biggest-files listing is the third site. An earlier revision of this pull request left it alone as display-only, which was wrong: `skills/assess/SKILL.md` documents that listing as part of the run summary the agent reads, so a tied boundary showed an agent a different top five on two runs of the same commit. It is fixed here. There are no tests for `render` and this change does not add a harness for one; the shared selection it now calls is tested directly instead, on a tied boundary with the input order reversed.

No document describes how either selection picks its files. `skills/assess/SKILL.md` states the hint's firing condition (the five largest are JSON, YAML or JSONL scored by scc with complexity 0) and `README.md` states the dominance threshold; neither names an order, so no doc goes stale.

## Risk Assessment

Low. The change is ordering only, confined to three sort keys inside one function. Files whose primary values differ keep their rank; only tied files move, and moving them is the fix. No schema change, no new key, no signature change to `write_stats`.

One visible effect: on a repository whose existing `.assess/complexity-stats.json` holds a tied group at the boundary, the first run after this change may report a hotspot as graduated and another as new. That is the one-off cost of settling on a stable order, not a regression.

## Deployment Notes

None. Plugin version 1.88.4; users pick it up through `/plugin update`.

## Known follow-ups

Not fixed here, recorded so it is not lost when this closes.

- `complexity-treemap.py`, `_keyhole_budget_rollup`: `over_subtrees` selects on `-est_tokens` alone over a dict whose insertion order is scanner emission order, and the result reaches `run-context.json`'s `stats_summary`, so an agent reads it in that order. Same defect class as the three lists, with the same `(-key, path)` fix. Its entries are subtrees rather than files, so it is outside this issue's scope, and ties among subtree token sums are rarer than ties on per-file complexity or line count.

### CODE REVIEW

File: .claude-plugin/plugin.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
-  "version": "1.88.1",
+  "version": "1.88.2",
```
[the remark this answers is not in the document]
Comment (bjcoombs) [reply .claude-plugin/plugin.json]: Fixed. Merged main (1.88.3 at 385cecb) and set the version to 1.88.4, so the bump no longer collides and the standalone-skills tag for this version has not been published. The number is set against main at merge time, not at spawn, because main advances while a pull request is open - this branch has now been overtaken twice on that line.

File: skills/assess/scripts/complexity-treemap.py [hunk 2]
```diff
@@ -535,6 +541,27 @@ def _warn_if_dominated_by_one_file(
...
+    Both callers pick "the biggest few" off the same file list and differ only
+    in what they do when a path carries no token estimate, so ``size_of`` is
+    passed in rather than branched on here. Ties break on the path, ascending,
+    under the plain byte ordering the top-10 lists use: without it the set
+    followed scanner emission order, so a tied boundary could show a different
+    few on two runs of the same commit - silencing the scc-only hint on one run
+    and not the next, and changing the biggest-files summary an agent reads
+    (issue #426). Every path shares the scan root, so ordering on the absolute
+    posix path is the same order as on the repository-relative one, and neither
+    caller is given the root.
```
[the remark this answers is not in the document]
Comment (bjcoombs) [reply skills/assess/scripts/complexity-treemap.py]: Fixed in 53df894. You are right - render takes root at line 639 and uses it at 716, so the clause was false. Deleted it and kept the one the conclusion actually rests on: every path shares the scan root, so ordering on the absolute posix path is the same order as on the repository-relative one. One line, comment text only, no behaviour change.
